src/cryptonote_core/blockchain.cpp

```diff
@@ -2075,8 +2077,8 @@ bool Blockchain::find_blockchain_supplement(const uint64_t req_start_block, cons
 
   m_db->block_txn_start(true);
   total_height = get_current_blockchain_height();
-  size_t count = 0;
-  for(size_t i = start_height; i < total_height && count < max_count; i++, count++)
+  size_t count = 0, size = 0;
+  for(size_t i = start_height; i < total_height && count < max_count && (size < FIND_BLOCKCHAIN_SUPPLEMENT_MAX_SIZE || count < 3); i++, count++)
```

Comment:
Will this mess up callers that expect `max_count` to be reached in this situation? Presumably you wouldn't make this change if that was the case but its always worth asking ...

Reply:
This is just a max, AFAIK no callers expect that exact number (it could also be less if this is near the end of the chain, so they have to handle it anyway).